src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -1009,6 +1009,53 @@ public function testMirrorCopiesRelativeLinkedContents()
         $this->assertEquals('\\' === DIRECTORY_SEPARATOR ? realpath($sourcePath.'\nested') : 'nested', readlink($targetPath.DIRECTORY_SEPARATOR.'link1'));
     }
 
+    public function testMirrorContentsWithSameNameAsSourceOrTargetWithoutDeleteOption()
+    {
+        $sourcePath = $this->workspace.DIRECTORY_SEPARATOR.'source'.DIRECTORY_SEPARATOR;
+
+        $oldPath = getcwd();
```

Comment:
Could you give a reason please? The save/restore `chdir()`s are following the same pattern as the existing tests `testFilesExistsFails()` and `testMirrorCopiesRelativeLinkedContents()` and it seems good practice to leave things as you find them.